Bluetooth: host: Fix PAST dependency on connections

cvinayak opened the pull request:
Periodic Advertising Sync Transfer depends on connections
hence enclose the LE Periodic Advertising Sync Transfer
Received Event code inside CONFIG_BT_CONN conditional
compilation.

Fixes #30120

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: host: Fix Periodic Adv Set Info Transfer dependency

Periodic Advertising Set Info Transfer depends on Extended
Advertising and connections hence enclose the LE Periodic
Advertising Set Info Sync Transfer implementation inside
CONFIG_BT_PER_ADV and CONFIG_BT_CONN conditional
compilation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: host: Fix PAST dependency on connections

Periodic Advertising Sync Transfer depends on connections
hence enclose the LE Periodic Advertising Sync Transfer
Received Event code inside CONFIG_BT_CONN conditional
compilation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (changes requested):
Huh, I thought I had already done that. Might have been missing from one of the merge conflicts I had for the other PA stuff.

Requesting change:  Lines starting from `bt_le_per_adv_sync_transfer` to after `bt_le_per_adv_sync_transfer_unsubscribe` should also be guarded.

cvinayak committed:
tests: Bluetooth: shell: Fix PAST commands conditional compilation

Enclose PAST subscribe/unsubscribe test commands inside
CONFIG_BT_PER_ADV_SYNC and CONFIG_BT_CONN conditional
compilation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (approved):
LGTM
